Sources/MessagingInApp/Gist/Managers/MessageManager.swift

```diff
@@ -5,12 +5,81 @@ public enum GistMessageActions: String {
     case close = "gist://close"
 }
 
+/**
+ Class that implements the business logic for a inline message being displayed.
+ */
+class InlineMessageManager: MessageManager {}
+
+/**
+ Class that implements the business logic for a modal message being displayed.
+ */
+class ModalMessageManager: MessageManager {
```

Comment:
Why are we using inheritance here?
I understand there is common logic, but we can extract the common logic in its own utility type, and use it in two different implementations.
Inheritance obscures the side effect of doing a single change and can be hard to trace at times

Reply:
See [my latest update](https://github.com/customerio/customerio-ios/pull/717#issuecomment-2135841512).